### PR TITLE
Generate changelog to release

### PR DESCRIPTION
**Pull request by jaroslav-fedor-swi**
Find all commits in given PATH between two tags, and add it as release notes into release.

* Excludes merge commits
* handles changelog differently for alpha/standard releases

## Image:

image

data source:
```
git log "swo-k8s-collector-4.0.0..swo-k8s-collector-4.1.0-alpha.4" --pretty=format:"%s" -- src build  | grep -v Merge
Bump library/golang in /build/docker
NH-90671 Bump base image
Bump distroless/static-debian12 in /build/docker
Bump debian from 12.6 to 12.7 in /build/docker
```

## Helm

image

### CODE REVIEW

File: .github/workflows/buildAndDeploy.yml [hunk 1]
```diff
@@ -56,6 +56,7 @@ jobs:
   
   build_and_test_windows:
     runs-on: windows-2022
+    if: startsWith(github.ref, 'refs/tags/') && !contains(github.ref, 'swo-k8s-collector')
```
Comment (jaroslav-fedor-swi) [inline .github/workflows/buildAndDeploy.yml:59]: reverting this to original configuration as it takes ~15mins to run for every commit.

File: .github/workflows/buildAndDeploy.yml [hunk 2]
```diff
@@ -207,6 +213,9 @@ jobs:
           tag_name: ${{ github.ref }}  # Use the tag that triggered the workflow
           release_name: ${{ github.ref }}
           body: |
-            **Full Changelog**: https://github.com/solarwinds/swi-k8s-opentelemetry-collector/compare/${{ env.PREVIOUS_TAG }}...${{ env.CURRENT_TAG}}
+              ## Changed
```
Comment (gantrior) [inline .github/workflows/buildAndDeploy.yml:216]:
```suggestion
              ## What's Changed
```

**Review by gantrior — APPROVED** [review]
"Generate release notes" button in GitHub UI still produces a bit better change log as it contains also PR refs and author, like:
> * Fix service releases by @jaroslav-fedor-swi in https://github.com/solarwinds/swi-k8s-opentelemetry-collector/pull/717

but it is good enough

**Comment by jaroslav-fedor-swi** [comment]
We can add authors, and possibly found PR related to commit, however is that what we want? GH generate button include all changes even not related to the release. It works well if you have single purpose repo.

**Pull request merged**